src/backend/distributed/utils/shardinterval_utils.c

```diff
@@ -74,29 +74,31 @@ CompareShardIntervals(const void *leftElement, const void *rightElement,
 	/*
 	 * Left element should be treated as the greater element in case it doesn't
 	 * have min or max values.
-	 */
-	if (!leftShardInterval->minValueExists || !leftShardInterval->maxValueExists)
-	{
-		comparisonResult = 1;
-		return comparisonResult;
-	}
-
-	/*
 	 * Right element should be treated as the greater element in case it doesn't
 	 * have min or max values.
 	 */
-	if (!rightShardInterval->minValueExists || !rightShardInterval->maxValueExists)
+	if (!leftShardInterval->minValueExists || !leftShardInterval->maxValueExists ||
```

Comment:
I find it slightly hard to follow the conditions and the calculation. What about something like the following for ease of read:
```C
bool leftHasNull = (!leftShardInterval->minValueExists || !leftShardInterval->maxValueExists);
bool rightHasNull = (!rightShardInterval->minValueExists || !rightShardInterval->maxValueExists);

if (leftHasNull && rightHasNull)
{
    /* rely on shardId for order to get consistent results */
	comparisonResult = 0;
}
else if (leftHasNull)
{
	comparisonResult = 1;
}
else if (rightHasNull)
{
	comparisonResult = -1;
}
else
{
	/* if both shard interval have min/max values, calculate the comparison result */
	...

}

...
```